src/Arcus.Messaging.Pumps.ServiceBus/Extensions/MessageCorrelationInfoExtensions.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using Arcus.Messaging.Abstractions;
+using GuardNet;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Arcus.Messaging.Pumps.ServiceBus.Extensions
+{
+    /// <summary>
+    /// Azure ServiceBus plugin to add correlation information to the inbound and outbound messages.
+    /// </summary>
+    public static class MessageCorrelationInfoExtensions
+    {
+        /// <summary>
+        /// Append the <see cref="MessageCorrelationInfo"/> after the message is received.
+        /// </summary>
+        /// <param name="message">The received message.</param>
+        public static MessageCorrelationInfo GetCorrelationInfo(this Message message)
+        {
+            return GetCorrelationInfo(message, NullLogger.Instance);
+        }
+
+        /// <summary>
+        /// Append the <see cref="MessageCorrelationInfo"/> after the message is received.
+        /// </summary>
+        /// <param name="message">The received message.</param>
+        /// <param name="logger">The logger to write diagnostic trace messages while constructing the correlation from the message's information.</param>
+        public static MessageCorrelationInfo GetCorrelationInfo(this Message message, ILogger logger)
+        {
+            Guard.NotNull(message, nameof(message));
+            logger ??= NullLogger.Instance;
+
+            string transactionId = DetermineTransactionId(message);
+            string operationId = DetermineOperationId(message.CorrelationId, logger);
+
+            var messageCorrelationInfo = new MessageCorrelationInfo(transactionId, operationId);
+            logger.LogInformation(
```

Comment:
Is this the responsibility of the extension or the user? I'd say the latter

Reply:
Yes,, I thought so too; 
I maybe even put the other log also somewhere else.